Mathlib/Analysis/Polynomial/MahlerMeasure.lean

```diff
@@ -0,0 +1,126 @@
+/-
+Copyright (c) 2025 Fabrizio Barroero. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fabrizio Barroero
+-/
+
+import Mathlib.Analysis.Analytic.Polynomial
+import Mathlib.Analysis.CStarAlgebra.Classes
+import Mathlib.Analysis.SpecialFunctions.Integrability.LogMeromorphic
+import Mathlib.MeasureTheory.Integral.CircleAverage
+
+/-!
+# Mahler measure of complex polynomials
+
+In this file we define the Mahler measure of a polynomial over `ℂ[X]` and prove some basic
+properties.
+
+## Main definitions
+
+- `Polynomial.logMahlerMeasure p`: the logarithmic Mahler measure of a polynomial `p` defined as
+`(2 * π)⁻¹ * ∫ x ∈ (0, 2 * π), log ‖p (e ^ (i * x))‖`.
+- `Polynomial.mahlerMeasure p`: the (exponential) Mahler measure of a polynomial `p`, which is equal
+to `e ^ p.logMahlerMeasure` if `p` is nonzero, and `0` otherwise.
+
+## Main results
+
+- `Polynomial.mahlerMeasure_mul`: the Mahler measure of the product of two polynomials is the
+product of their Mahler measures.
+-/
+
+namespace Polynomial
+
+open Real
+
+/-- The logarithmic Mahler measure of a polynomial `p` defined as
+`(2 * π)⁻¹ * ∫ x ∈ (0, 2 * π), log ‖p (e ^ (i * x))‖` -/
+noncomputable def logMahlerMeasure (p : ℂ[X]) := circleAverage (fun x ↦ log ‖eval x p‖) 0 1
+
+theorem logMahlerMeasure_def (p : ℂ[X]) : p.logMahlerMeasure =
+    circleAverage (fun x ↦ log ‖eval x p‖) 0 1 := rfl
```

Comment:
```suggestion
theorem logMahlerMeasure_def (p : ℂ[X]) :
    p.logMahlerMeasure = circleAverage (fun x ↦ log ‖eval x p‖) 0 1 := rfl
```
Can you try to avoid splitting formula across lines whenever possible?